unstructured_ingest/v2/processes/connectors/redisdb.py

```diff
@@ -0,0 +1,209 @@
+import json
+import shutil
+from dataclasses import dataclass, field
+from pathlib import Path
+from typing import TYPE_CHECKING, Any, Optional
+
+from pydantic import Field, Secret
+
+from unstructured_ingest.error import DestinationConnectionError
+from unstructured_ingest.utils.data_prep import batch_generator
+from unstructured_ingest.utils.dep_check import requires_dependencies
+from unstructured_ingest.v2.interfaces import (
+    AccessConfig,
+    ConnectionConfig,
+    FileData,
+    Uploader,
+    UploaderConfig,
+    UploadStager,
+    UploadStagerConfig,
+)
+from unstructured_ingest.v2.logger import logger
+from unstructured_ingest.v2.processes.connector_registry import DestinationRegistryEntry
+
+if TYPE_CHECKING:
+    from redis.asyncio import Redis
+
+CONNECTOR_TYPE = "redis"
+SERVER_API_VERSION = "1"
+
+
+class RedisAccessConfig(AccessConfig):
+    uri: Optional[str] = Field(
+        default=None, description="If not anonymous, use this uri, if specified."
+    )
+    password: Optional[str] = Field(
+        default=None, description="If not anonymous, use this password, if specified."
+    )
+
+
+class RedisConnectionConfig(ConnectionConfig):
+    access_config: Secret[RedisAccessConfig] = Field(
+        default=RedisAccessConfig(), validate_default=True
+    )
+    host: Optional[str] = Field(
+        default=None, description="hostname or IP address of a single redis instance to connect to"
+    )
+    database: Optional[int] = Field(default=0, description="database index to connect to")
+    port: Optional[int] = Field(default=6379)
+    username: Optional[str] = Field(default=None)
+    ssl: Optional[bool] = Field(default=True)
+    connector_type: str = Field(default=CONNECTOR_TYPE, init=False)
+
+
+class RedisUploadStagerConfig(UploadStagerConfig):
+    pass
+
+
+@dataclass
+class RedisUploadStager(UploadStager):
+    upload_stager_config: RedisUploadStagerConfig = field(
+        default_factory=lambda: RedisUploadStagerConfig()
+    )
+
+    def run(
+        self,
+        elements_filepath: Path,
+        file_data: FileData,
+        output_dir: Path,
+        output_filename: str,
+        **kwargs: Any,
+    ) -> Path:
+        output_path = Path(output_dir) / Path(f"{output_filename}.json")
+        shutil.copy(elements_filepath, output_path)
+        return output_path
```

Comment:
We don't need to implement stager if it doesn't actually do any transformations.
For reference [local](https://github.com/Unstructured-IO/unstructured-ingest/blob/0dc113ca963fe543297bb22a50c0e2f0e029c2d2/unstructured_ingest/v2/processes/connectors/local.py#L204) and [fsspec](https://github.com/Unstructured-IO/unstructured-ingest/blob/main/unstructured_ingest/v2/processes/connectors/fsspec/fsspec.py) include just `Uploader` implementations.